client/src/js/base/InitialIcon.js

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import styled from "styled-components";
+import { getColor, getFontWeight, theme } from "../app/theme";
+
+export const StyledInitialIcon = styled.span`
+    border-radius: 50%;
+    display: flex;
+    flex: 0 0 auto;
+    justify-content: center;
+    align-items: center;
+    height: ${props => `${props.size}px`};
+    width: ${props => `${props.size}px`};
+    background: ${props => `hsl(${props.hash}, 83%, 21%);`}
+    color: ${getColor({ color: "white", theme })};
+    font-weight: ${getFontWeight("thick")};
+    font-size: ${props => `${props.size / 1.95}px`};
+`;
+
+export const InitialIcon = ({ handle, size }) => {
+    let hash = 0;
+    for (let i = 0; i < handle.length; i++) {
+        hash = (hash << 5) - handle.charCodeAt(i);
+        hash = hash & hash;
+    }
```

Comment:
Fixed. I am used useMemo instead of use callback so that I could also calculate the mod only once, but otherwise it should be functionally identical to use callback. This highlighted code has now been reduced to two lines by using reduce.